packages/cloud_firestore/cloud_firestore/example/test_driver/load_bundle_e2e.dart

```diff
@@ -0,0 +1,104 @@
+      return Uint8List.fromList(string.codeUnits);
+    }
+
+    setUp(() {
+      firestore = FirebaseFirestore.instance;
+    });
+
+    group('FirebaseFirestore.loadBundle()', () {
+      test('loadBundle()', () async {
+        Uint8List buffer = await loadBundleSetup();
+        LoadBundleTask task = firestore.loadBundle(buffer);
+
+        // ensure the bundle has been completely cached
+        await task.stream().last;
+
+        QuerySnapshot<Map<String, Object?>> snapshot = await firestore
+            .collection(collection)
+            .get(const GetOptions(source: Source.cache));
+
+        expect(
+          snapshot.docs.map((document) => document['number']),
+          everyElement(anyOf(1, 2, 3)),
+        );
+      });
+
+      test('loadBundle(): LoadBundleTaskProgress stream snapshots', () async {
+        Uint8List buffer = await loadBundleSetup();
+        LoadBundleTask task = firestore.loadBundle(buffer);
+
+        final list = await task.stream().toList();
+
+        expect(list.map((e) => e.totalDocuments), everyElement(isNonNegative));
+        expect(list.map((e) => e.bytesLoaded), everyElement(isNonNegative));
+        expect(list.map((e) => e.documentsLoaded), everyElement(isNonNegative));
+        expect(list.map((e) => e.totalBytes), everyElement(isNonNegative));
+        expect(list, everyElement(isInstanceOf<LoadBundleTaskSnapshot>()));
+
+        LoadBundleTaskSnapshot lastSnapshot = list.removeLast();
+        expect(lastSnapshot.taskState, LoadBundleTaskState.success);
+
+        expect(
+          list.map((e) => e.taskState),
+          everyElement(LoadBundleTaskState.running),
+        );
+      });
+    });
+
+    group('FirebaeFirestore.namedQueryGet()', () {
+      test('namedQueryGet() successful', () async {
+        Uint8List buffer = await loadBundleSetup();
+        LoadBundleTask task = firestore.loadBundle(buffer);
+
+        // ensure the bundle has been completely cached
+        await task.stream().last;
+
+        // namedQuery 'named-bundle-test' which returns a QuerySnaphot of the same 3 documents
+        // with 'number' property
+        QuerySnapshot<Map<String, Object?>> snapshot =
+            await firestore.namedQueryGet('named-bundle-test',
+                options: const GetOptions(source: Source.cache));
+
+        expect(snapshot.docs.map((document) => document['number']),
+            everyElement(anyOf(1, 2, 3)));
+      });
+
+      test('namedQueryGet() error', () async {
+        Uint8List buffer = await loadBundleSetup();
+        LoadBundleTask task = firestore.loadBundle(buffer);
+
+        // ensure the bundle has been completely cached
+        await task.stream().last;
+
+        await expectLater(
+            firestore.namedQueryGet('wrong-name',
+                options: const GetOptions(source: Source.cache)),
+            throwsA(isA<FirebaseException>().having((e) => e.message, 'message',
+                contains('Named query has not been found'))));
```

Comment:
```suggestion
        await expectLater(
          firestore.namedQueryGet(
            'wrong-name',
            options: const GetOptions(source: Source.cache),
          ),
          throwsA(
            isA<FirebaseException>().having((e) => e.message, 'message',
                contains('Named query has not been found')),
          ),
        );
```

(reformatted by hand, likely will need a `dart format`